connectivity/check/deployment.go

```diff
@@ -235,6 +243,141 @@ func (ct *ConnectivityTest) deploy(ctx context.Context) error {
 		}
 	}
 
+	// For performance workloads, we want to ensure the client/server are in the same zone
+	// If a zone has > 1 node, use that zone
+	zones := map[string]int{}
+	zone := ""
+	lz := ""
+	n, err := ct.client.ListNodes(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to query nodes")
+	}
+	for _, l := range n.Items {
+		if _, ok := zones[l.GetLabels()["topology.kubernetes.io/zone"]]; ok {
+			zone = l.GetLabels()["topology.kubernetes.io/zone"]
+			break
+		} else {
+			zones[l.GetLabels()["topology.kubernetes.io/zone"]] = 1
+			lz = l.GetLabels()["topology.kubernetes.io/zone"]
+		}
+	}
+	// No zone had > 1, use the last zone.
+	if zone == "" {
+		ct.Debug("Each zone only has a single node - could impact the performance test results")
+		zone = lz
+	}
+
+	// Need to capture the IP of the Server Deployment, and pass to the client to execute benchmark
+	_, err = ct.clients.src.GetDeployment(ctx, ct.params.TestNamespace, PerfClientDeploymentName, metav1.GetOptions{})
+	if err != nil {
+		ct.Logf("✨ [%s] Deploying Perf Client deployment...", ct.clients.src.ClusterName())
+		perfClientDeployment := newDeployment(deploymentParameters{
+			Name:  PerfClientDeploymentName,
+			Kind:  kindPerfName,
+			Port:  80,
+			Image: defaults.ConnectivityPerformanceImage,
+			Labels: map[string]string{
+				"client": "role",
+			},
+			Command: []string{"/bin/bash", "-c", "sleep 10000000"},
+			Affinity: &corev1.Affinity{
+				NodeAffinity: &corev1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
```

Comment:
To circumvent the above(changing the current logic to determine zone when it exists), I could change this to Preferred.